### PR TITLE
cacheSelfie.py

### PR DESCRIPTION
**Pull request by hssahota2**

### CODE REVIEW

File: python/selfie-lib/selfie_lib/Selfie.py [hunk 1]
```diff
@@ -1,8 +1,29 @@
...
+
+
+system_instance: Union[SnapshotSystem, None] = None
+
+
+def get_system() -> SnapshotSystem:
+    global system_instance
+    if system_instance is None:
+        system_instance = _selfieSystem()
+    return system_instance
```
Comment (nedtwigg) [inline python/selfie-lib/selfie_lib/Selfie.py:26]: We already have this, `_selfieSystem()`, look at line 34 just below this

File: python/selfie-lib/selfie_lib/Selfie.py [hunk 2]
```diff
@@ -16,3 +37,18 @@ def expect_selfie(actual: Union[str, int, bool]):
         return BooleanSelfie(actual)
     else:
         raise NotImplementedError()
+
+
+def cache_selfie_string(to_cache: Cacheable[str]) -> CacheSelfie[str]:
+    """Create a CacheSelfie instance for caching strings with identity transformation."""
+    identity_roundtrip = Roundtrip.identity()
+    deferred_disk_storage = get_system().disk_thread_local()
+    return CacheSelfie(deferred_disk_storage, identity_roundtrip, to_cache)
```
Comment (nedtwigg) [inline python/selfie-lib/selfie_lib/Selfie.py:46]: This should call `cache_selfie_generic`.

**Review by nedtwigg — CHANGES_REQUESTED** [review]
Very close! Have you tested it in our example project?

A good example would be something like:

```python
import random

cache_selfie(lambda: str(random.random())).to_be_TODO()
```

It should only update once, but when you add `_TODO` or `#selfieonce` it should rewrite and then stop updating.

**Comment by hssahota2** [comment]
I tried testing the cache selfie but right now I'm getting an assertion error

**Review by nedtwigg — APPROVED** [review]
Works great!

**Pull request merged**